include/deal.II/lac/diagonal_matrix.h

```diff
@@ -179,6 +179,12 @@ class DiagonalMatrix : public Subscriptor
                    const VectorType &src) const;
 
   /**
+   * Initialize vector @p dst. This is a part of the interface required
+   * by linear operator.
```

Comment:
do you mean `linear_operator()`? If you write it that way, it'll show up linked in doxygen.